### ISSUE TITLE
Field with same starting name in mongoose model schema is ignored

### ISSUE THREAD

**Issue by IamAfnanSk** [francescov1/mongoose-tsgen#70]
```TypeScript
const GitHubInfo = new mongoose.Schema({
	login: String,
	id: Number,
	node_id: String
})

const UserSchema = new mongoose.Schema(
	{
		name: { type: String, required: true },
		github: GitHubInfo,
		githubProfileURL: { type: String, default: '' },
		twitterProfileURL: { type: String, default: '' }
	},
	{ collection: 'users' }
)
```

In this schema there are 2 fields gihub and githubProfileURL but after generating types with mongoose-tsgen, githubProfileURL is skipped and all other fields are typed in the output file.

**Comment by francescov1** [comment]
@IamAfnanSk Realized this actually broke a specific case where the generator handles nested properties (need to consider keys like `city.address.country`, which isnt handled by your fix. I created a second PR for this, can you verify if your issue is fixed by this change? https://github.com/francescov1/mongoose-tsgen/pull/72/files#diff-0031f10c8304cb81d02e83ba1b26c8f9273cc02fea992e27df58ea2abc63074eR205

**Comment by IamAfnanSk** [comment]
@francescov1 Hey! I didn't thought about that.
Also yes PR #72 works with this issue's condition too. Lets' publish the new version. 💪🏼

**Comment by francescov1** [comment]
Published in 8.4.2 🚀
